modules/thermal_hydraulics/src/components/HeatStructureCylindrical.C

```diff
@@ -21,6 +21,14 @@ HeatStructureCylindrical::validParams()
   params.addRequiredParam<std::vector<unsigned int>>("n_part_elems",
                                                      "Number of elements of each radial region");
   params.addParam<std::vector<std::string>>("materials", "Material name for each radial region");
+  params.addParam<std::vector<UserObjectName>>(
+      "solid_properties", "Solid properties object name for each radial region");
+  params.addParam<std::vector<Real>>(
+      "solid_properties_T_ref",
+      {},
+      "Density reference temperatures for each radial region. This is required if "
+      "'solid_properties' is provided. The density in each region will be a constant value "
+      "computed by evaluating the density function at the reference temperature.");
```

Comment:
We cannot put this in `HeatStructureBase` because it would not work with the `HeatStructure3DFromFile`, right?

Reply:
Right, we could add some kind of support in 3D heat structures later as well, but here they are tied to the 2D heat structure layers.